bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/FindReplaceLogic.java

```diff
@@ -588,12 +588,11 @@ public void updateTarget(IFindReplaceTarget newTarget, boolean canEditTarget) {
 		this.isTargetEditable = canEditTarget;
 
 		if (this.target != newTarget) {
-			if (newTarget != null && newTarget instanceof IFindReplaceTargetExtension)
-				((IFindReplaceTargetExtension) newTarget).endSession();
+			if (this.target instanceof IFindReplaceTargetExtension)
```

Comment:
We should end the session for the current target not for the new target.

Reply:
Thank you for finding this, @merks! This is a regression from #1132. Your proposal properly restored the previous behavior.